Create self-contained  module build

alistairjcbrown opened the pull request:
The previous PR, which aimed to fix requiring this component without a transpile step, was unsuccessful - https://github.com/ubilabs/react-geosuggest/pull/5

A transpiled, but non-packed distribution is required. The current distribution files include global variables  and wiring for use of `require` where it is not supported. This conflicts with a build system which supports require and not global variables.

The existing build has been set up solely for browserify output. This PR adds a new gulp task `build-module`, which uses `babel` to transpile the files into plain Javascript, but without packing the files together. These are output into the `react-geosuggest.module` directory within `dist` and the `package.json` has been updated to point to that directory. An `index.js` file is created as the entry point.

ro-ka commented:
Thanks for you PR!

For the additional changes – can you split them as own PRs to not mix up and have a descriptive history?

I’m not sure if checking in the `react-geosuggest.module` is the way to go… It is only needed for an `npm publish`. Maybe we should build and then publish? Doing this in [another project](https://github.com/ubilabs/node-geobatch/blob/master/package.json) and it is working fine. Need to check whether the `react-component-gulp-tasks` have this. Also I was thinking on switching to npm scripts. What do you think?

ro-ka commented:
As we switched to Travis CI @Ubilabs, I configured it here, too. Only checking eslint with our code styles. A merge of the current master is appreciated. :)

alistairjcbrown commented:
@ro-ka I've just rebased against master. I'll look at splitting out PRs and updating this tonight.

> For the additional changes – can you split them as own PRs
Will do

> I’m not sure if checking in the `react-geosuggest.module` is the way to go
Fair enough - I'm not sure how you manage your build files and publish.

> I was thinking on switching to npm scripts
I'm a big fan of using `npm run x` as it abstracts away what the underlying tool is and means that I don't have to install it globally :)

alistairjcbrown commented:
I've split off the additional changes and updated the description for this PR to reflect.
New PRS are #9 and #10

I've also left off the built files that I'd included before; I'll leave you to decide
a) if this is the best approach
b) if you want to generate the build files as a separate commit, on publish, etc.

@ro-ka While I remember - I don't think your `dist` directory is included in your npm publish at the moment; when I do an `npm install` and check the output `react-geosuggest` directory, it's missing.

ro-ka commented:
Thanks for the update! I will review this later – still thinking about switching to pure npm based tasks.

ro-ka commented:
Removed gulp and switched to npm scripts.
